celesta-sql/src/main/java/ru/curs/celesta/score/discovery/ScoreByScoreResourceDiscovery.java

```diff
@@ -0,0 +1,88 @@
+package ru.curs.celesta.score.discovery;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Enumeration;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+
+import ru.curs.celesta.CelestaException;
+import ru.curs.celesta.score.io.Resource;
+import ru.curs.celesta.score.io.UrlResource;
+
+/**
+ * Implementation of score discovery based on JAR-files resources look up.
+ */
+public final class ScoreByScoreResourceDiscovery implements ScoreDiscovery {
+
+    private static final String SCORE_FILES_LOCATION = "score/score.files";
+
+    @Override
+    public Set<Resource> discoverScore() {
+
+        Map<String, Resource> grainNameToResourceMap = new LinkedHashMap<>();
+
+        ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+        try {
+            Enumeration<URL> urls = (classLoader != null)
+                    ? classLoader.getResources(SCORE_FILES_LOCATION)
+                    : ClassLoader.getSystemResources(SCORE_FILES_LOCATION);
+
+            while (urls.hasMoreElements()) {
+                Resource scoreFilesResource = new UrlResource(urls.nextElement());
+
+                InputStream scoreFilesInputStream;
+                try {
+                    scoreFilesInputStream = scoreFilesResource.getInputStream();
+                } catch (IOException ex) {
+                    System.out.println("score index file is missing: " + scoreFilesResource.toString());
+                    continue;
+                }
+
+                try (BufferedReader reader = new BufferedReader(
+                        new InputStreamReader(scoreFilesInputStream, StandardCharsets.UTF_8))) {
+                    Iterable<String> li = reader.lines()::iterator;
```

Comment:
This looks fancy. Never knew it can be done this way 😃 
But how about more conservative way to read lines from `BufferedReader`?

```java
String gp;
while ((gp = reader.readLine()) != null) {
```

Reply:
Yes indeed slipped out of my sight.